api/src/main/java/org/openmrs/module/pihcore/htmlformentry/action/AddPatientToQueueAction.java

```diff
@@ -37,65 +36,68 @@ public class AddPatientToQueueAction implements CustomFormSubmissionAction {
+                    }
+                    queue = queues.get(0);
 
-                List<Concept> allowedStatuses = queueServicesWrapper.getAllowedStatuses(queue);
-                if (!allowedStatuses.isEmpty()) {
-                    queueEntry.setStatus(allowedStatuses.get(0));
-                }
+                    queueEntry = new QueueEntry();
+                    queueEntry.setQueue(queue);
+                    queueEntry.setPatient(patient);
 
-                List<Concept> allowedPriorities = queueServicesWrapper.getAllowedPriorities(queue);
-                if (!allowedPriorities.isEmpty()) {
-                    queueEntry.setPriority(allowedPriorities.get(0));
-                }
-                queueEntry.setStartedAt(encounter.getEncounterDatetime());
-                queueEntry.setVisit(encounter.getVisit());
+                    List<Concept> allowedStatuses = queueServicesWrapper.getAllowedStatuses(queue);
+                    if (!allowedStatuses.isEmpty()) {
+                        queueEntry.setStatus(allowedStatuses.get(0));
+                    }
 
-                QueueEntrySearchCriteria searchCriteria = new QueueEntrySearchCriteria();
-                searchCriteria.setPatient(patient);
-                searchCriteria.setQueues(Collections.singletonList(queue));
-                List<QueueEntry> queueEntries = queueServicesWrapper.getQueueEntryService().getQueueEntries(searchCriteria);
-                boolean isPatientAlreadyOnTheQueue = false;
-                for (QueueEntry entry : queueEntries) {
-                    if (QueueUtils.datesOverlap(entry.getStartedAt(), entry.getEndedAt(), queueEntry.getStartedAt(), queueEntry.getEndedAt())) {
-                        isPatientAlreadyOnTheQueue = true;
-                        break;
+                    List<Concept> allowedPriorities = queueServicesWrapper.getAllowedPriorities(queue);
+                    if (!allowedPriorities.isEmpty()) {
+                        queueEntry.setPriority(allowedPriorities.get(0));
                     }
+                    queueEntry.setStartedAt(encounter.getEncounterDatetime());
+                    queueEntry.setVisit(encounter.getVisit());
 
-                }
-                if (!isPatientAlreadyOnTheQueue) {
-                    try {
-                        queueServicesWrapper.getQueueEntryService().saveQueueEntry(queueEntry);
-                    } catch (Exception e) {
-                        throw new RuntimeException("AddPatientToQueueAction failed to add patient to queue", e);
+                    QueueEntrySearchCriteria searchCriteria = new QueueEntrySearchCriteria();
+                    searchCriteria.setPatient(patient);
+                    searchCriteria.setQueues(Collections.singletonList(queue));
+                    searchCriteria.setIsEnded(null);
+                    List<QueueEntry> queueEntries = queueServicesWrapper.getQueueEntryService().getQueueEntries(searchCriteria);
```

Comment:
This is a nit, but it seems odd to me that you instantiate the new QueryEntry and fill out all the parameter *before* actually doing the check to see if you need to save it.  Not really a big deal, but maybe add a comment here "// only save queue entry if patient not on queue" just for readability... I was a little confused because after line 67 would have assumed that the decision to save the queue had already been made, and a comment would highlight that it hadn't.  Hope that makes some sense!

Reply:
Thanks @mogoodrich ! I have refactored the code, and moved the code you pointed out in the same block as saving the new queue entry.